managed/CounterStrikeSharp.API/Modules/Menu/IMenu.cs

```diff
@@ -0,0 +1,44 @@
+﻿/*
+ *  This file is part of CounterStrikeSharp.
+ *  CounterStrikeSharp is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  CounterStrikeSharp is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with CounterStrikeSharp.  If not, see <https://www.gnu.org/licenses/>. *
+ */
+
+using System.Collections.Generic;
+
+namespace CounterStrikeSharp.API.Modules.Menu
+{
+    public interface IMenu
+    {
+        public string Title { get; set; }
+        public List<ChatMenuOption> MenuOptions { get; }
+        
+        public ChatMenuOption AddMenuOption(string display, Action<CCSPlayerController, ChatMenuOption> onSelect, bool disabled = false);
+    }
+
+    public interface IMenuInstance
+    {
+        protected IMenu Menu { get; }
+        protected CCSPlayerController? Player { get; }
+        protected int Page { get; }
+        protected int CurrentOffset { get; }
+        protected int NumPerPage { get; }
+        protected Stack<int> PrevPageOffsets { get; }
+        
+        public void NextPage();
+        public void PrevPage();
+        public void Reset();
+        public void Display();
+        public void OnKeyPress(CCSPlayerController player, int key);
```

Comment:
While this technically isn't _wrong_, I would recommend renaming this function to something else (maybe `OnOptionSelected`) to avoid confusion between this implementation of menus and the old SourceMod style of menus. There currently isn't a way to recreate those menus or to detect key presses. Maybe when we get support for UserMessages in the future 🤞

Reply:
It’s still called OnKeyPress to maintain backwards compatibility, plus once we figure out how to detect the number keys it’ll make sense and won’t need refactoring.